src/Hl7.Fhir.Base/Specification/Terminology/MultiTerminologyService.cs

```diff
@@ -15,19 +15,85 @@ namespace Hl7.Fhir.Specification.Terminology
+        public MultiTerminologyService(IEnumerable<ITerminologyService> services) : this(services.Select(s => TerminologyServiceRoutingSettings.CreateDefault(s))) { }
+
+        /// <summary>
+        /// A collection of multiple terminology services to allow for one or multiple fallback services when validating codes for example.
+        /// </summary>
+        /// <param name="settings">Routing settings to be used. You can add settings to prefer certain terminology services, the others will still be used based as fallback based on their order.</param>
+        public MultiTerminologyService(IEnumerable<TerminologyServiceRoutingSettings> settings)
         {
-            if (services == null) throw Error.ArgumentNull(nameof(services));
-            _services.AddRange(services);
+            if (settings == null || !settings.Any()) throw Error.ArgumentNull(nameof(settings));
+
+            int count = 0;
+            foreach (var setting in settings)
+            {
+                _termServices.Add(count, setting);
+                count++;
+            }
+        }
+
+        /// <summary>
+        /// A collection of multiple terminology services to allow for one or multiple fallback services when validating codes for example.
+        /// </summary>
+        /// <param name="services">Terminology services to be used. Note that the first service in the list will be used first, the others will be used for fallback based on their order.</param>
+        public MultiTerminologyService(params ITerminologyService[] services) : this(services.ToList()) { }
+
+
+        /// <summary>
+        /// A collection of multiple terminology services to allow for one or multiple fallback services when validating codes for example.
+        /// </summary>
+        /// <param name="services">Orderable terminology services to be used. You can set the order of the services to be used using the settings, the others will be used for fallback based on their order.</param>
+        public MultiTerminologyService(params TerminologyServiceRoutingSettings[] services) : this(services.ToList()) { }
+
+
+        /// <summary>
+        /// Add an <see cref="ITerminologyService"/> at the back of the order of terminologyServices to be used.
+        /// </summary>
+        /// <param name="service"><see cref="ITerminologyService"/> to be added.</param>
+        public void Add(ITerminologyService service)
+        {
+            var settings = new TerminologyServiceRoutingSettings(service);
+            Add(settings);
+        }
+
+        /// <summary>
+        /// Add an <see cref="ITerminologyService"/> including their routing settings at the back of the order of terminologyServices to be used.
+        /// </summary>
+        /// <param name="settings"><see cref="TerminologyServiceRoutingSettings"/> to be added.</param>
+        public void Add(TerminologyServiceRoutingSettings settings)
+        {
+            var position = _termServices.Keys.Last() + 1;
```

Comment:
Since the order is now based on a automatically assigned sequential number and it is not possible to specifity a prio via the settings, the Keys in the OrderedLst will always just be 1...N.  This means you could also use a normal List now, and Insert() at the beginning or Add at the end!